streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableReduceTest.java

```diff
@@ -78,4 +92,118 @@ public void shouldAddAndSubtract() {
         strings.addAll(right);
         return strings;
     }
+
+    @Test
+    public void stringSetSerdeTest() {
+        final Set<String> originalSet = new HashSet<>();
+        final String topicName = "serdes";
+
+        final String string1 = "One";
+        final String string2 = "Two";
+        final String string3 = "Three";
+        originalSet.add(string1);
+        originalSet.add(string2);
+        originalSet.add(string3);
+
+        final byte[] bytes = setSerde.serializer().serialize(topicName, originalSet);
+        final Set<String> newSet = setSerde.deserializer().deserialize(topicName, bytes);
+
+        assertTrue(newSet.containsAll(originalSet));
+        assertEquals(originalSet.size(), newSet.size());
+    }
+
+    private static final class StringSetSerde extends WrapperSerde<Set<String>> {
+        public StringSetSerde() {
+            super(new StringSetSerializer(), new StringSetDeserializer());
+        }
+
+        private static class StringSetSerializer implements Serializer<Set<String>> {
+            Serde<String> stringSerde = Serdes.String();
+            Serde<Integer> intSerde = Serdes.Integer();
+
+            @Override
+            public void configure(final Map<String, ?> configs, final boolean isKey) {
+            }
+
+            @Override
+            public byte[] serialize(final String topic, final Set<String> data) {
+                if (data == null)
+                    return null;
+
+                final List<byte[]> bytesList = new LinkedList<>();
+                int totalBytes = 0;
+
+                // Data is encoded as 4 bytes containing N, the length in bytes of the next string, followed by the N bytes containing that string
+                for (final String string : data) {
+                    final byte[] stringBytes = stringSerde.serializer().serialize(topic, string);
+                    final int numStringBytes = stringBytes.length;
+                    final byte[] lengthBytes = intSerde.serializer().serialize(topic, numStringBytes);
+
+                    totalBytes += numStringBytes + 4;
+                    bytesList.add(lengthBytes);
+                    bytesList.add(stringBytes);
+                }
+
+                // now that we know the total number of bytes needed we can allocate an array of that size and copy the individual byte[] over
+                final byte[] serializedBytes = new byte[totalBytes];
+
+                int i = 0;
+                for (final byte[] bytes : bytesList) {
+                    for (final byte b : bytes) {
+                        serializedBytes[i++] = b;
+                    }
+                }
+                return serializedBytes;
+            }
+
+            @Override
+            public void close() {
+                // nothing to do
```

Comment:
we should call `stringSerde.close()` and `intSerde.close()` here (in case we don't remove this code anyway)